ingress/controller.go

```diff
@@ -45,7 +45,11 @@ func (c *controller) Run() {
 	}
 
 	// initialize with current state of ingress
-	c.updateLoadBalancer()
+	err = c.updateLoadBalancer()
```

Comment:
We should definitely mutex protect the loadbalancer if it needs it. Which it probably does, since separate goroutines will be accessing it.

Doing the initialisation after creating a watch is the only way to ensure we don't miss an update. Otherwise we may miss an update between the initial select and the subsequent subscription. I don't see any race condition here, the watcher will simply call getIngresses if there is an update.

We could also expose the resource version directly on the client, that would avoid all problems, and it wouldn't matter what order we call things. But it is leaking more kubernetes logic into the controller. And we'll have to return resource version from `getIngresses`.